lib/prefix.c

```diff
@@ -1017,3 +1017,120 @@ inet6_ntoa (struct in6_addr addr)
   inet_ntop (AF_INET6, &addr, buf, INET6_ADDRSTRLEN);
   return buf;
 }
+
+
+static uint8_t convertchartohexa(uint8_t * hexa, int *error)
+{
+  if ((*hexa == '0') || (*hexa == '1') || (*hexa == '2') ||
+      (*hexa == '3') || (*hexa == '4') || (*hexa == '5') ||
+      (*hexa == '6') || (*hexa == '7') || (*hexa == '8')
+      || (*hexa == '9'))
+    return (uint8_t) (*hexa) - '0';
+  if ((*hexa == 'a') || (*hexa == 'A'))
+    return 0xa;
+  if ((*hexa == 'b') || (*hexa == 'B'))
+    return 0xb;
+  if ((*hexa == 'c') || (*hexa == 'C'))
+    return 0xc;
+  if ((*hexa == 'd') || (*hexa == 'D'))
+    return 0xd;
+  if ((*hexa == 'e') || (*hexa == 'E'))
+    return 0xe;
+  if ((*hexa == 'f') || (*hexa == 'F'))
+    return 0xf;
+  *error = -1;
+  return 0;
+}
+
+/* converts to internal representation of mac address
+ * returns 1 on success, 0 otherwise 
+ * format accepted: AA:BB:CC:DD:EE:FF
+ * if mac parameter is null, then check only
+ */
+int str2mac(const char *str, char *mac)
```

Comment:
what happens if user gives "Aa:Bd:43:56:66". 
The function checks for lower and upper cases characters.
Is your function doing the same ?